lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -2434,5 +2434,124 @@ def test_cell_method_correct_order(self):
         self.assertTrue(cube1 == cube2)
 
 
+class Test__summary_dim_name(tests.IrisTest):
+    def test_standard_named_dim(self):
+        cube = Cube(np.zeros((5, 3)))
+        cube.add_dim_coord(
+            DimCoord(np.arange(5), standard_name="longitude", units="degrees"),
+            0,
+        )
+        result = cube._summary_dim_name(0)
+        self.assertEqual(result, "longitude")
+
+    def test_long_named_dim(self):
+        cube = Cube(np.zeros((5, 3)))
+        cube.add_dim_coord(
+            DimCoord(np.arange(5), long_name="alons", units="degrees"), 0
+        )
+        result = cube._summary_dim_name(0)
+        self.assertEqual(result, "alons")
+
+    def test_unmapped_dim(self):
+        cube = Cube(np.zeros((5, 3)))
+        cube.add_dim_coord(DimCoord(np.arange(5), long_name="x", units=1), 0)
+        result = cube._summary_dim_name(1)
+        self.assertEqual(result, "-- ")
+
+    def test_aux_mapped_dim(self):
+        # It should *not* treat an aux-coord as the name of a dim.
+        cube = Cube(np.zeros((5, 3)))
+        cube.add_aux_coord(
+            DimCoord(np.arange(5), standard_name="longitude", units="degrees"),
+            0,
+        )
+        result = cube._summary_dim_name(0)
+        self.assertEqual(result, "-- ")
+
+
+class Test__summary_vector_sections_info(tests.IrisTest):
+    def test_section_titles(self):
+        cube = Cube([0, 1])
+        result = cube._summary_vector_sections_info()
+        self.assertEqual(len(result), 5)
+        self.assertEqual(
+            [sect[0] for sect in result],
+            [
+                "Dimension coordinates",
+                "Auxiliary coordinates",
+                "Derived coordinates",
+                "Cell measures",
+                "Ancillary variables",
+            ],
+        )
+
+    def test_extra_lines(self):
+        cube = Cube([0, 1])
+        result = cube._summary_vector_sections_info()
+        self.assertEqual(len(result), 5)
+        self.assertEqual(
+            [sect[2] for sect in result], [True, True, True, False, False]
+        )
+
+    def test_dim_aux_coords(self):
+        cube = Cube(np.zeros((3, 5, 4)))
+        cube.add_dim_coord(
+            DimCoord(np.arange(3), long_name="lats", units="degrees"), 0
+        )
+        cube.add_aux_coord(
+            DimCoord(np.arange(5), standard_name="longitude", units="degrees"),
+            1,
+        )
```

Comment:
After this: add a second, almost identical instance with an alphabetically earlier name to test the [sorting behaviour](https://github.com/SciTools/iris/pull/3914/files#diff-38eaa350bf479c486e9f6659b8077be2d52448f17739872e50285b98b5423415R2355-R2357).